applications/present3D/present3D.cpp

```diff
@@ -805,8 +805,6 @@ int main( int argc, char **argv )
         viewer.realize();
 
         if (doSetViewer) setViewer(viewer, width, height, distance);
-
-        viewerInitialized = true;
     }
```

Comment:
Removing this line is safe with the current rev of the code but it's bad practice given how similar code elsewhere in the main relies upon the viewInitilized = true being executed, without it it would introduce a bug.  The danger in removing this currently redundant operation if  order of code changes slightly or this block is copied an pasted earlier suddenly we'd introduce a bug. 

Finally if (!initialized) { do initialization; initialized = true; } is a common technique, the pattern is balanced, it helps tell the programmer what the code does and when it's complete the job of initialization is done.  Without this any programmer reviewing the code will be left wondering is there anything else to be initialized.

All in all cppcheck in this instance is technically correct from a pure C++ -> machine code basis, but wrong form the perspective that C++ code is a live and breathing record of what the programmer intends for the computer to do, and a part of the dialogue with future programmers reviewing the code.

Reply:
I see your point and agree. Will take this into account for next PR's.